addons/crm_livechat/models/crm_lead.py

```diff
@@ -0,0 +1,28 @@
+from ast import literal_eval
+
+from odoo import _, fields, models
+from odoo.exceptions import UserError
+
+
+class CrmLead(models.Model):
+    _inherit = "crm.lead"
+
+    channel_id = fields.Many2one(
+        "discuss.channel", "Live chat from which the lead was created", index="btree_not_null"
+    )
+
+    def write(self, vals):
+        if "channel_id" in vals:
```

Comment:
Would instead check current has read access on channel before allowing to change it, that way we check there is no access escalation, if I understood the feature. No need to be that restrictive.